src/main/java/nextstep/subway/domain/Sections.java

```diff
@@ -3,61 +3,126 @@
+        .filter(registeredSection -> registeredSection.containSameDownStation(section))
+        .findFirst();
 
-        sections.add(section);
+    if (upMatch.isPresent()) {
+      Section registeredSection = upMatch.get();
+      compareDistance(section, registeredSection);
+      sections.add(
+          Section.of(line, registeredSection.getDistance() - section.getDistance(),
+              section.getDownStation(), registeredSection.getDownStation()));
+      sections.remove(registeredSection);
+    }
+    if (downMatch.isPresent()) {
+      Section registeredSection = downMatch.get();
+      compareDistance(section, registeredSection);
+      sections.add(
+          Section.of(line, registeredSection.getDistance() - section.getDistance(),
+              registeredSection.getUpStation(), section.getUpStation()));
+      sections.remove(registeredSection);
     }
+    sections.add(section);
+  }
 
-    public void deleteSection(Station downStation) {
-        if (sections.stream().count() <= 1) {
-            throw new IllegalStateException("Section에 삭제할 수 있는 Station이 없습니다.");
-        }
-        Section lastSection = getLastSection();
-        if (!lastSection.getDownStation().equals(downStation)) {
-            throw new IllegalStateException("Section에 삭제할 수 있는 Station이 없습니다.");
-        }
-        getStations().stream().filter(station -> station.equals(downStation)).findAny().ifPresent(x -> new IllegalStateException("Section에 삭제할 수 있는 Station이 없습니다."));
-        sections.remove(sections.size()-1);
+  private void alreadyRegisteredValidate(Section registeredSection) {
+    sections.stream().filter(section -> !section.containsSameStations(registeredSection))
+        .findFirst().orElseThrow(() -> new IllegalArgumentException("동일한 Section이 존재합니다."));
+  }
+
+  private void notRegisteredValidate(Section registeredSection) {
+    if (!getStations().contains(registeredSection.getUpStation()) && !getStations().contains(
+        registeredSection.getDownStation())) {
+      throw new IllegalArgumentException("등록된 Station이 해당 노선에 존재하지 않습니다.");
     }
+  }
 
-    public List<Station> getStations() {
-        if (sections.isEmpty()) {
-            return Collections.emptyList();
-        }
-        List<Station> stationList = new ArrayList<>();
-        stationList.add(sections.get(0).getUpStation());
-        sections.forEach(section -> stationList.add(section.getDownStation()));
+  public void deleteSection(Station downStation) {
+    if (sections.stream().count() <= 1) {
+      throw new IllegalStateException("Section에 삭제할 수 있는 Station이 없습니다.");
+    }
+    Section lastSection = getLastSection();
+    if (!lastSection.getDownStation().equals(downStation)) {
+      throw new IllegalStateException("Section에 삭제할 수 있는 Station이 없습니다.");
+    }
+    getStations().stream().filter(station -> station.equals(downStation)).findAny()
+        .ifPresent(x -> new IllegalStateException("Section에 삭제할 수 있는 Station이 없습니다."));
```

Comment:
```suggestion
.ifPresent(x -> {
    throw new IllegalStateException("Section에 삭제할 수 있는 Station이 없습니다.");
  });
```

이렇게 해주셔야 예외가 던져질거에요 😄

Reply:
흐미... 
compile 에러도 뜨지 않았네요...